### PR TITLE
Document `__table args__`

### PR DESCRIPTION
**Pull request by hammerhead**
## Summary of the changes / Why this is an improvement
As per the title.

## Checklist

 - [X] [CLA](https://crate.io/community/contribute/cla/) is signed

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -73,7 +73,7 @@ def read(path):
               'createcoverage>=1,<2',
               'stopit>=1.1.2,<2',
               'flake8>=4,<5'],
-        doc=['sphinx>=3,<4',
+        doc=['sphinx>=3.5,<6',
```
Comment (hammerhead) [inline setup.py:76]: Python 3.10 support was added in Sphinx 4.3.0, so Sphinx in this project didn't run for me. This updates the version constraint to be aligned with the latest crate-docs-theme:

https://github.com/crate/crate-docs-theme/blob/66b054f08d0eab4c809a86a420c707c56dd7271e/setup.py#L61

Comment (amotl) [reply setup.py:76]: I am fine with relaxing the upper bound. What about this change?
```suggestion
        doc=['sphinx>=3,<6',
```

Comment (amotl) [reply setup.py:76]: Oh, same as `crate-docs-theme`. All right!

**Review by amotl — APPROVED** [review]
Thank you!

**Pull request merged**